app/src/fdroid/java/com/eventyay/organizer/core/attendee/ScanQRViewModel.java

```diff
@@ -0,0 +1,196 @@
+package com.eventyay.organizer.core.attendee;
+
+import androidx.annotation.VisibleForTesting;
+import androidx.lifecycle.LiveData;
+import androidx.lifecycle.ViewModel;
+
+import com.eventyay.organizer.R;
+import com.eventyay.organizer.common.Constants;
+import com.eventyay.organizer.common.ContextManager;
+import com.eventyay.organizer.common.livedata.SingleEventLiveData;
+import com.eventyay.organizer.common.rx.Logger;
+import com.eventyay.organizer.data.Preferences;
+import com.eventyay.organizer.data.attendee.Attendee;
+import com.eventyay.organizer.data.attendee.AttendeeRepository;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.concurrent.TimeUnit;
+
+import javax.inject.Inject;
+
+import io.reactivex.Observable;
+import io.reactivex.disposables.CompositeDisposable;
+import io.reactivex.subjects.PublishSubject;
+
+import static com.eventyay.organizer.common.rx.ViewTransformers.schedule;
+
+public class ScanQRViewModel extends ViewModel {
```

Comment:
Why not reusing it?

Reply:
Opened a PR #1706 for this. It needs to be converted to ViewModel first so that it can be reused. But currently #1706 is not working properly, I'll fix it.

Reply:
@iamareebjamal Is it feasible to use the same ViewModel for both?

```
public void onBarcodeDetected(Barcode barcode) {
        detect.onNext(barcode == null);

        if (barcode == null || attendees.isEmpty())
            return;

        data.onNext(barcode.displayValue);
    }
```

This would require to import `com.google.android.gms.vision.barcode.Barcode` which cannot be there in the F-Droid build variant.

Reply:
Create a wrapper class for barcode in both flavors. Or extract this dependent code in separate component for both flavors. Whichever feels easier. Tell me if you need more assistance

Reply:
@iamareebjamal Please have a look at #1706 

I have updated accordingly and it's working, is it the correct way of doing it?

Reply:
Please remind again if I forget tomorrow

Reply:
Okay

Reply:
@iamareebjamal Please check.